lib/matplotlib/pyplot.py

```diff
@@ -543,9 +531,26 @@ def make_active(event):
         _pylab_helpers.Gcf.set_active(figManager)
         figManager.canvas.figure.number = num
 
+        if _INSTALL_FIG_OBSERVER:
+            figManager.canvas.figure.add_callback(_auto_draw_if_interactive)
+
     return figManager.canvas.figure
 
 
+def _auto_draw_if_interactive(fig):
+    """
+    This is an internal helper function for making sure that auto-redrawing
+    works as intended in the plain python repl.
+
+    Parameters
+    ----------
+    fig : Figure
+        A figure object which is assumed to be associated with a canvas
+    """
+    if fig.stale and matplotlib.is_interactive():
```

Comment:
"does not get updates" = "cannot update"?  Well if we assume that, then we just add an unnecessary check....

However consider tools, if we add a tool that calls ``my_line2d.set_color('purple')``, this line means you need to add a ``canvas.draw_idle()`` after that in the tool, but if you run in interactive mode, you don't need it.  This feels like inconsistent behaviour... but happy to get persuaded why we don't/may not want that.